examples/connext_dds/CMakeLists.txt

```diff
@@ -160,26 +184,38 @@ add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/flat_data_api/c++98")
 add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/flat_data_latency/c++11")
 
 # fragmented_data_statistics
-add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/fragmented_data_statistics/c")
-add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/fragmented_data_statistics/c++")
-add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/fragmented_data_statistics/c++11")
+if(${RTICONNEXTDDS_VERSION} VERSION_GREATER_EQUAL "6.1.0")
+    add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/fragmented_data_statistics/c")
+    add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/fragmented_data_statistics/c++")
+    add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/fragmented_data_statistics/c++11")
+else()
+    send_skip_message("fragmented_data_statistics")
+endif()
 
 # get_publishers
 add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/get_publishers/c")
```

Comment:
in this case, we do not check `if(${RTICONNEXTDDS_VERSION} VERSION_GREATER_EQUAL "6.0.0")` because we use the find_package with the version 6.0.0.
I'm not a cmake expert, but is it better to have `if(${RTICONNEXTDDS_VERSION} VERSION_GREATER_EQUAL "6.0.0")` for these examples, or keep it as it is? (@lulivi @ManuelJNunez )
Also, in the case that we add the checking for RTI Connext DDS 6.0.0, we can do it by groups:
```
if(${RTICONNEXTDDS_VERSION} VERSION_GREATER_EQUAL "6.0.0")
 #add all the examples of 6.0.0
else()
 #log warning
endif()

if(${RTICONNEXTDDS_VERSION} VERSION_GREATER_EQUAL "6.1.0")
 #add all the examples of 6.1.0
else()
 #log warning
endif()
```

Reply:
@angelrti The problem is maintaining that file if someone forgets to update it. Now it's ordered in alphabetical order to facilitate the maintaining. Maybe we can update the [pull request template](https://github.com/rticommunity/rticonnextdds-examples/blob/develop/.github/PULL_REQUEST_TEMPLATE.md) and add this:

```markdown
<!-- Uncomment bellow if you added a C/C++ example and updated examples/connext_dds/CMakeList.txt -->
- [ ] I added a new C/C++ example and updated `resources/connext_dds/CMakeList.txt`
```

By doing so, we will avoid forgetting to update the top-level `CMakeList.txt`.

Reply:
sounds good

Reply:
conclusions per our discussion about this topic:
We will have one `if` clause per version and all the examples that require that version as minimum Connext version will live under that if. For example:
```
if(${RTICONNEXTDDS_VERSION} VERSION_GREATER_EQUAL "6.1.0")
    add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/detect_samples_dropped/c")
    add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/detect_samples_dropped/c++")
    add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/detect_samples_dropped/c++11")
add_subdirectory(
        "${CMAKE_CURRENT_SOURCE_DIR}/dynamic_data_access_union_discriminator/c")
    add_subdirectory(
        "${CMAKE_CURRENT_SOURCE_DIR}/dynamic_data_access_union_discriminator/c++")
    add_subdirectory(
        "${CMAKE_CURRENT_SOURCE_DIR}/dynamic_data_access_union_discriminator/c++11")
...
else()
    #probably list all the examples here that have been skipped
endif()
```
This will help with maintainability when we have more version-specific examples in the future.

Then, we will create (a different issue) a macro to introspect the language folders per example:
```
add_example(detect_samples_dropped)
add_example(dynamic_data_access_union_discriminator)
```
Avoiding adding all the `example_name/language`